precompiles/gov/query_test.go

```diff
@@ -35,35 +35,67 @@ func (s *PrecompileTestSuite) TestGetVotes() {
 	var ctx sdk.Context
 	method := s.precompile.Methods[gov.GetVotesMethod]
 	testCases := []struct {
-		name     string
-		malleate func() []gov.WeightedVote
-		args     []interface{}
-		expPass  bool
-		expTotal uint64
-		gas      uint64
+		name        string
+		malleate    func() []gov.WeightedVote
+		args        []interface{}
+		expPass     bool
+		errContains string
+		expTotal    uint64
+		gas         uint64
 	}{
 		{
 			name: "valid query",
 			malleate: func() []gov.WeightedVote {
-				err := s.network.App.GovKeeper.AddVote(s.network.GetContext(), 1, s.keyring.GetAccAddr(0), []*govv1.WeightedVoteOption{{Option: govv1.OptionYes, Weight: "1.0"}}, "")
-				s.Require().NoError(err)
-				return []gov.WeightedVote{
-					{ProposalId: 1, Voter: s.keyring.GetAddr(0), Options: []gov.WeightedVoteOption{{Option: uint8(govv1.OptionYes), Weight: "1.0"}}},
+				proposalID := uint64(1)
+				voter := s.keyring.GetAccAddr(0)
+				voteOption := &govv1.WeightedVoteOption{
+					Option: govv1.OptionYes,
+					Weight: "1.0",
 				}
+
+				err := s.network.App.GovKeeper.AddVote(
+					s.network.GetContext(),
+					proposalID,
+					voter,
+					[]*govv1.WeightedVoteOption{voteOption},
+					"",
+				)
+				s.Require().NoError(err)
+
+				return []gov.WeightedVote{{
+					ProposalId: proposalID,
+					Voter:      s.keyring.GetAddr(0),
+					Options: []gov.WeightedVoteOption{
+						{
+							Option: uint8(voteOption.Option), //nolint:gosec // G115 -- integer overflow is not happening here
+							Weight: voteOption.Weight,
+						},
+					},
+				}}
 			},
 			args:     []interface{}{uint64(1), query.PageRequest{Limit: 10, CountTotal: true}},
 			expPass:  true,
 			expTotal: 1,
 			gas:      200_000,
 		},
 		{
-			name:    "invalid proposal ID",
-			args:    []interface{}{uint64(0), query.PageRequest{Limit: 10, CountTotal: true}},
-			expPass: false,
-			gas:     200_000,
-			malleate: func() []gov.WeightedVote {
-				return []gov.WeightedVote{}
-			},
+			name:        "invalid proposal ID",
+			args:        []interface{}{uint64(0), query.PageRequest{Limit: 10, CountTotal: true}},
+			expPass:     false,
+			gas:         200_000,
+			errContains: "proposal id can not be 0",
+		},
+		{
+			name:        "fail - invalid number of args",
+			args:        []interface{}{},
+			errContains: fmt.Sprintf(cmn.ErrInvalidNumberOfArgs, 2, 0),
+			gas:         200_000,
```

Comment:
we're always defining the same amount of gas in all tests, we could remove that from the test cases